### PR TITLE
fix: add typescript definition for this utility

### PR DESCRIPTION
**Pull request by val-patreon**
Internally, Patreon is trying its best to use TypeScript everywhere. However, that's made harder when some of its public  legacy utilities that it imports are not typed.

Relevant functions that I used to derive the type: https://github.com/Patreon/url-factory/blob/b7c275513e0eb1a8ba6a89512979bc298b427292/src/index.js#L6-L43

### CODE REVIEW

**Comment by mathewlee11** [comment]
TIL this repo exists and I get email notifications for it

**Comment by anaulin** [comment]
Same, @mathewlee11, same. Turns out https://github.com/watching is our friend.

**Comment by mathewlee11** [comment]
@anaulin that is much easier than my current strategy of unsubscribing from repos after the first email!

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+type ReturnInnerFuction = (relativeUrl: string, params?: Record<string, unknown>) => string
+
+export default function urlFactory(baseUrl?: string, baseParams?: Record<string, unknown>): ReturnInnerFuction;
```
Comment (dorwin-patreon) [inline index.d.ts]: the change itself looks good but you should add the `types` key and extend the `files` key in package.json ([example](https://github.com/Patreon/nion/blob/master/package.json#L7-L14))

**Review by dorwin-patreon — CHANGES_REQUESTED** [review]

File: types/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "jsx": "react",
```
Comment (dorwin-patreon) [inline types/tsconfig.json]: not needed since no react here?

**Review by dorwin-patreon — APPROVED** [review]

**Pull request merged**